seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/dag/physical/SubPlan.java

```diff
@@ -245,12 +244,10 @@ public void cancelPipeline() {
         }
         // If an active Master Node done and another Master Node active, we can not know whether canceled pipeline
         // complete. So we need cancel running pipeline again.
-        if (PipelineState.CANCELING.equals((PipelineState) runningJobStateIMap.get(pipelineLocation))) {
-            LOGGER.info(String.format("%s already in state CANCELING, skip cancel", pipelineFullName));
-        } else {
+        if (!PipelineState.CANCELING.equals((PipelineState) runningJobStateIMap.get(pipelineLocation))) {
```

Comment:
Is there a data type conversion exception? Or should declare: `IMap<String, PipelineState> runningJobStateIMap`

Reply:
> 

Under normal circumstances, no conversion exception will occur.